addons/mail/static/tests/discuss_app/discuss.test.js

```diff
@@ -2190,3 +2190,57 @@ test("Read of unread chat where new message is deleted should mark as read [REQU
         contains: [".badge", { count: 0 }],
     });
 });
+
+test("Deleting a message decreases the message unread counter", async () => {
+    const pyEnv = await startServer();
+    const bobUserId = pyEnv["res.users"].create({ name: "bob" });
+    const bobPartnerId = pyEnv["res.partner"].create({ name: "bob", user_ids: [bobUserId] });
+    const channelId = pyEnv["discuss.channel"].create({
+        channel_member_ids: [
+            Command.create({ partner_id: serverState.partnerId }),
+            Command.create({ partner_id: bobPartnerId }),
+        ],
+        channel_type: "chat",
+    });
+    const [memberId] = pyEnv["discuss.channel.member"].search([
+        ["channel_id", "=", channelId],
+        ["partner_id", "=", serverState.partnerId],
+    ]);
+    const messageId = pyEnv["mail.message"].create({
+        author_id: bobPartnerId,
+        body: "Hey!",
+        model: "discuss.channel",
+        res_id: channelId,
+        message_type: "comment",
+    });
+    pyEnv["discuss.channel.member"].write([memberId], {
+        seen_message_id: messageId,
+        message_unread_counter: 0,
+    });
+    await start();
+    await openDiscuss();
+    await withUser(bobUserId, () =>
+        rpc("/mail/message/post", {
+            post_data: { body: "Regrettable message", message_type: "comment" },
+            thread_id: channelId,
+            thread_model: "discuss.channel",
+        })
+    );
```

Comment:
I would cautiously wait for Discuss rendering with messages being done before letting bob post a message.

We are aware of some design problems in HOOT, notably that there's only 1 user being authenticated with the server at once. That means marking conversation as read while bob is posting a message would inadvertently make bob mark the conversation as read, instead of Mitchell Admin.

Reply:
> wait for Discuss rendering with messages being done

I'm not sure I know what this means/how to do it

Reply:
`await contains(".o-mail-Message")`, with the setup of thread containing at least 1 message.
This gives more confidence that RPCs during opening/loading of discuss happens with HOOT mock server user authenticated as Mitchell Admin rather than Bob

Reply:
In this case opening the thread would mess with the unread messages counter, perhaps I can add a message to the inbox thread just to wait the rendering of the message there?